### DISCUSSION TITLE
How do I enable support for wgsl?

### DISCUSSION THREAD

**Discussion by SKyletoft** [emacs-lsp/lsp-mode#4369] [Q&A] [answered]
So there's a config for wgsl-analyzer in the repo, and I've confirmed that it's also there in the version I have installed. It doesn't work by default with wgsl-mode in the same way other languages do and if I register it myself in my local config the configuration doesn't work and it just runs everything set to default (which is only syntax errors).

If I try to require it myself with a `(require 'lsp-wgsl)` I get a undefined symbol error.

i've looked through all the PRs that mention it and none of them mention anything extra you need to do to enable it and I can't find anything to suggest it's supposed to be disabled by default.

What am I missing?

I'm runnning Emacs 30 from nixpkgs and lsp-mode from nix's emacs-overlay (I think it just sources latest Melpa?) as of a few hours ago

**Comment by yyoncho** [comment]
> (require 'lsp-wgsl)

That should work although you don't need it. M-x lsp should be enough. Make sure that your local installation has that file. You may use `M-x find-library RET lsp-mode` and then look in the containing folder for that file.

**Reply by SKyletoft** [reply]
The file is there, it just doesn't load. If I open it and do eval-buffer it works, but not if I `require` it

**Reply by SKyletoft** [reply]
And in `*Messages*` I get the
```
LSP :: There are no language servers supporting current mode `wgsl-mode' registered with `lsp-mode'.
This issue might be caused by:
1. The language you are trying to use does not have built-in support in `lsp-mode'. You must install the required support manually. Examples of this are `lsp-java' or `lsp-metals'.
2. The language server that you expect to run is not configured to run for major mode `wgsl-mode'. You may check that by checking the `:major-modes' that are passed to `lsp-register-client'.
3. `lsp-mode' doesn't have any integration for the language behind `wgsl-mode'. Refer to https://emacs-lsp.github.io/lsp-mode/page/languages and https://langserver.org/ .
4. You are over `tramp'. In this case follow https://emacs-lsp.github.io/lsp-mode/page/remote/.
5. You have disabled the `lsp-mode' clients for that file. (Check `lsp-enabled-clients' and `lsp-disabled-clients').
You can customize `lsp-warn-no-matched-clients' to disable this message.
```
error.

I'm also getting some error from yasnippet. Can that be interfering?
```
Ignoring unknown directive "desc" in file: /nix/store/qk3ibgl4j0dgi1150jqzsr8syddw06fk-emacs-packages-deps/share/emacs/site-lisp/elpa/yasnippet-snippets-20240221.1621/snippets/c++-mode/class11
Ignoring unknown directive "\\${[0-9N]*" in file: /nix/store/qk3ibgl4j0dgi1150jqzsr8syddw06fk-emacs-packages-deps/share/emacs/site-lisp/elpa/yasnippet-snippets-20240221.1621/snippets/c++-mode/class11
Ignoring unknown directive "References" in file: /nix/store/qk3ibgl4j0dgi1150jqzsr8syddw06fk-emacs-packages-deps/share/emacs/site-lisp/elpa/yasnippet-snippets-20240221.1621/snippets/c++-mode/class11```

**Reply by SKyletoft** [reply]
Nope, disabling yasnippet entirely does nothing

**Reply by yyoncho** [reply]
Are you able to find it via M-x find-library?

**Reply by SKyletoft** [reply]
Yes

**Reply by SKyletoft** [reply]
https://youtu.be/57wsGDqqURs

This is what it looks like. It just isn't being loaded automatically. It's findable with M-x find-library and if I do that and load it with eval-buffer it all just works. It just doesn't do so by default

**Reply by yyoncho** [reply, marked as answer]
@SKyletoft your issue should be gone after https://github.com/emacs-lsp/lsp-mode/commit/d3aaa9d6b8521a9e6cf622dc37e662e9459720b6 . We were missing the code to autoload that package.

**Reply by SKyletoft** [reply]
Then I just have to wait for it to hit the repos in a few hours, thanks
